stdlib/public/core/ArrayBuffer.swift

```diff
@@ -567,6 +571,78 @@ extension _ArrayBuffer {
     }
   }
 
+  @inlinable @_alwaysEmitIntoClient
+  static var associationKey: UnsafeRawPointer {
+    //We never dereference this, we just need an address to use as a unique key
+    UnsafeRawPointer(Builtin.addressof(&_swiftEmptyArrayStorage))
+  }
+  
+  @inlinable @_alwaysEmitIntoClient
+  internal func getAssociatedBuffer() -> _ContiguousArrayBuffer<Element>? {
+    let getter = objc_getAssociatedObject as @convention(c)(
+      Any,
+      UnsafeRawPointer
+    ) -> Any?
+    let typedGetter = unsafeBitCast(
+      getter,
+      to: (@convention(c)(
+        AnyObject,
+        UnsafeRawPointer
+      ) -> AnyObject?).self
+    )
+    if let assoc = typedGetter(
+      _storage.objCInstance,
+      _ArrayBuffer.associationKey
+    ) {
+      let buffer = unsafeBitCast(
+        assoc,
+        to: _ContiguousArrayStorage<Element>.self
+      )
+      return _ContiguousArrayBuffer(buffer)
+    }
+    return nil
+  }
+  
+  @inlinable @_alwaysEmitIntoClient
+  internal func setAssociatedBuffer(_ buffer: _ContiguousArrayBuffer<Element>) {
+    let setter = unsafeBitCast(getSetAssociatedObjectPtr(), to: (@convention(c)(
+      AnyObject,
+      UnsafeRawPointer,
+      AnyObject?,
+      UInt
+    ) -> Void).self)
+    setter(
+      _storage.objCInstance,
+      _ArrayBuffer.associationKey,
+      buffer._storage,
+      0o1401 //OBJC_ASSOCIATION_RETAIN
+    )
+  }
+  
+  @_alwaysEmitIntoClient @inline(never)
+  internal func withUnsafeBufferPointer_nonNative<R>(
+    _ body: (UnsafeBufferPointer<Element>) throws -> R
+  ) rethrows -> R {
+    objc_sync_enter(_storage.objCInstance)
+    var associatedBuffer = getAssociatedBuffer()
+    let unwrapped: _ContiguousArrayBuffer<Element>
+    if let associatedBuffer {
+      unwrapped = associatedBuffer
+    } else {
+      associatedBuffer = ContiguousArray(self)._buffer
+      unwrapped = associatedBuffer.unsafelyUnwrapped
+      setAssociatedBuffer(unwrapped)
+    }
+    defer { _fixLifetime(unwrapped) }
+    objc_sync_exit(_storage.objCInstance)
```

Comment:
Pretty sure this is true but just checking: this is definitely 100% sure always going to be the exact same pointer that you got from `_storage.objCInstance` above?